sanic/__main__.py

```diff
@@ -157,12 +222,20 @@ def main():
         else:
             ssl = None
 
+        if args.fast:
+            try:
+                args.workers = len(os.sched_getaffinity(0))
+            except AttributeError:
+                error_logger.warning(
+                    "Ignoring '--fast' since it is not supported on this OS."
+                )
+
```

Comment:
Does this exclude MacOS from using `--fast`? I cannot test on Mac right now but I recall someone saying that this is specific to Linux.

Reply:
My mac (or rather, my wife's mac) is being repaired now so I cannot test. The docs just say "UNIX".

> These functions control how a process is allocated CPU time by the operating system. They are only available on some Unix platforms. For more detailed information, consult your Unix manpages.

Reply:
I'll pull and test - I have both intel and arm (m1) macs available to work with.

Reply:
So on MacOS on an m1, the tests pretty much result in fails across the board. @ahopkins it may make sense to work with you to get a runner set up on my m1 mac mini that can run the tox tests and we're at least tracking the rate of failure, that way if someone gets a bee in their bonnet to fixup things for MacOS they have a starting point.

Reply:
@Tronic - the answer is yes:
```
$ uname -a && python
Darwin goldfinger.local 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:20 PDT 2021; root:xnu-7195.141.6~3/RELEASE_ARM64_T8101 arm6
Python 3.9.7 (default, Sep  3 2021, 04:31:11) 
[Clang 12.0.5 (clang-1205.0.22.9)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>> import os
>>> len(os.sched_getaffinity(0))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'os' has no attribute 'sched_getaffinity'
```

Reply:
So I suggest adding a fallback to "normal" cpu count if this function is not available. OS detection or special handling of Windows might not be necessary, because there is a limit to one worker elsewhere (app.py IIRC).

Reply:
Adding `cpu_count()` as the fallback.